ilastik/widgets/hdf5SubvolumeSelectionDialog.py

```diff
@@ -18,50 +18,78 @@
     QDialog,
+    QDialogButtonBox,
     QLabel,
     QLineEdit,
+    QListWidget,
     QTextEdit,
     QVBoxLayout,
     QWidget,
 )
 
-from PyQt5.QtCore import Qt
-
 from lazyflow.utility import globList
 
 
 class SubvolumeSelectionDlg(QDialog):
-    """
-    A window to ask the user to choose between multiple HDF5 datasets in a single file.
+    """A window to ask the user to choose between multiple HDF5 datasets in a single file.
+
+    If multi is True, user can select multiple datasets at once.
     """
 
-    def __init__(self, datasetNames, parent):
+    def __init__(self, datasetNames, parent, *, multi=False):
         super().__init__(parent)
-        label = QLabel(
-            "Your HDF5/N5 File contains multiple image volumes.\n" "Please select the one you would like to open."
-        )
-
-        self.combo = QComboBox()
-        for name in datasetNames:
-            self.combo.addItem(name)
-
-        buttonbox = QDialogButtonBox(Qt.Horizontal, parent=self)
-        buttonbox.setStandardButtons(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
-        buttonbox.accepted.connect(self.accept)
-        buttonbox.rejected.connect(self.reject)
+        self.multi = multi
 
         layout = QVBoxLayout()
-        layout.addWidget(label)
-        layout.addWidget(self.combo)
-        layout.addWidget(buttonbox)
-
         self.setLayout(layout)
 
+        label_lines = ["<p>Your file contains multiple images.</p>"]
+        if multi:
+            label_lines.append("<p>Select images that you would like to open.</p>")
+        else:
+            label_lines.append("<p>Select image that you would like to open.</p>")
+        layout.addWidget(QLabel("".join(label_lines), parent=self))
+
+        if multi:
+            self.items = QListWidget(parent=self)
+            self.items.setSelectionMode(QListWidget.MultiSelection)
+            self.items.insertItems(0, datasetNames)
+            layout.addWidget(self.items)
+        else:
+            self.combo = QComboBox(parent=self)
+            self.combo.addItems(datasetNames)
+            layout.addWidget(self.combo)
+
+        buttons = QDialogButtonBox(Qt.Horizontal, parent=self)
+        buttons.setStandardButtons(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        buttons.accepted.connect(self.accept)
+        buttons.rejected.connect(self.reject)
+        layout.addWidget(buttons)
+
+    def select_path(self) -> Optional[str]:
+        """Select and return an item text, or None if the dialog has not been accepted."""
+        if self.multi:
+            raise ValueError("selectedPath can be used only when multi=False")
+        if self.exec_() != self.Accepted:
+            return None
+        return self.combo.currentText()
+
+    def select_paths(self) -> Optional[List[str]]:
+        """Select and return item texts, or None if the dialog has not been accepted."""
+        if not self.multi:
+            raise ValueError("selectedPaths can be used only when multi=True")
```

Comment:
not sure something that doesn't take a parameter should raise `ValueError`. From [the docs](https://docs.python.org/3/library/exceptions.html#ValueError):

> Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.